### PR TITLE
Timer Driver + onboard LED driver + project unification 

### PR DESCRIPTION
**Pull request by akashkollipara**
- Porting timer driver for FE310 with this commit.
- Adding uinfied interface for all timer calls
- Add onboardLED driver
- Enable project unification

### CODE REVIEW

**Comment by akashkollipara** [comment]
Folks please review carefully.
This change is relatively very large and many files are changed with interlinked changes.

File: src/platform/mega_avr/common/arch/panic.S [hunk 1]
```diff
@@ -13,4 +13,4 @@
 function plat_panic_handler
 	push	r24
 	ldi	r24, 254
-	rjmp	isr
+	jmp	isr
```
Comment (shreeroopajay) [inline src/platform/mega_avr/common/arch/panic.S:16]: any specific reason for using jmp instead of rjmp ? jmp uses an extra cy for fetching addr

**Review by shreeroopajay — COMMENTED** [review]
some suggestions/questions from performance perspective

File: src/platform/mega_avr/common/arch/panic.S (hunk 1 above)
Comment (akashkollipara) [reply src/platform/mega_avr/common/arch/panic.S:16]: As compilation goes through, this function falls beyond range of rjmp there by causing linker to fail address association.

File: src/lib/libc/time.c [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		delay = 1;
+	else
+		delay = d;
+
+	ret = get_timestamp(&tstamp);
+	delay += tstamp;
+	do
+	{
+		arch_wfi();
+		ret = get_timestamp(&tstamp);
```
[the remark this answers is not in the document]
Comment (akashkollipara) [reply src/lib/libc/time.c:47]: We have minimum time resolution of 1ms as per timer irq.

File: src/lib/libc/time.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ */
+status_t udelay(uint16_t d)
+{
+	status_t ret;
+	uint64_t tstamp;
+	uint64_t delay;
+
+	if(d > 2000)
+		return error_func_inval_arg;
+	else if(d == 0)
```
[the remark this answers is not in the document]
Comment (akashkollipara) [reply src/lib/libc/time.c:37]: Not a usual case but it's better to have a trap I feel.

**Review by pranjalchanda08 — APPROVED** [review]

**Pull request merged**